sql/log_event.cc

```diff
@@ -1906,16 +1906,18 @@ Query_log_event::dummy_event(String *packet, ulong ev_offset,
 }
 
 /*
-  Replace an event (GTID event) with a BEGIN query event, to be compatible
+  Replace a Gtid_log_event with a BEGIN query event, to be compatible
   with an old slave.
 */
 int
 Query_log_event::begin_event(String *packet, ulong ev_offset,
+                             Format_description_log_event* fdev,
                              enum_binlog_checksum_alg checksum_alg)
 {
   uchar *p= (uchar *)packet->ptr() + ev_offset;
   uchar *q= p + LOG_EVENT_HEADER_LEN;
   size_t data_len= packet->length() - ev_offset;
+  Gtid_log_event original_event= Gtid_log_event(p, data_len, fdev);
```

Comment:
`packet->length()` is a `uint32`, `ev_offset` is a `ulong`, their difference is stashed into `size_t 
 data_len` and it’s then passed to `Gtid_log_event(const uchar, uint, const Format_description_log_event*)`.
[What is this madness?](https://jira.mariadb.org/browse/MDEV-26896)